### DISCUSSION TITLE
Support for uploading list of Devices?

### DISCUSSION THREAD

**Discussion by gaby** [seriousm4x/UpSnap#803] [Feature requests]
I have a weird use-case where there's around 50-70 hosts, but the lab uses a `/16`. Using the built-in scan in UpSnap is not feasable, I just takes forever and generates a lot of noise in the network with the large number of IP's. 

Currently I could add devices one by one, but it would be useful to be able to add multiple from a file of mac addresses. Once upload UpSnap would scan each MAC to get information about IP's, or Domains, etc.

**Comment by gaby** [comment]
@seriousm4x Thoughts?

**Comment by seriousm4x** [comment]
I like the idea of giving the user multiple options to import devices automatically. I thought about a new section here called "Import" or something next to network scan.
![image](https://github.com/user-attachments/assets/862e569a-4893-4e71-9151-213f4bc8544a)
And then giving the user multiple options to batch import devices. Just some ideas:
- Text field with mac addresses splitted by line (basically what you want)
- Upload a json
- ... ?

**Reply by gaby** [reply]
@seriousm4x Yes, that would work. Just another tab with a different way of importing devices. 💪

**Reply by gaby** [reply]
Endline delimited file, or json, or yaml would work.

I believe only the MAC Address should/could be the required field. The UI then can have the same options to specify things like the ones in `Manual` or `Network Scan`.

**Reply by gaby** [reply]
@seriousm4x Was there an update on adding this feature?

**Reply by seriousm4x** [reply]
Not yet, thanks for reminding me. I plan it for the next release.

**Reply by gaby** [reply]
Thank you! 💪

**Reply by seriousm4x** [reply]
Ok, so there is an issue I just stumbled upon. Following scenario:

1. Paste a bunch of mac addresses into a text box and send it to the backend
2. Backend needs to resolve the mac address to get the IP address

Problem: It's not possible to "scan a single mac address". To get the IP address from a mac you would need to scan a network range and then match the found mac's to the given ones. So you are back at this point:

> Using the built-in scan in UpSnap is not feasable, I just takes forever and generates a lot of noise in the network with the large number of IP's.

A possible solution would be to make both the mac and ip address required when batch importing. But that seems unintuitive and also is the website supposed to look like?

I haven't found a good solution to this yet.

**Reply by gaby** [reply]
> Ok, so there is an issue I just stumbled upon. Following scenario:
> 
> 1. Paste a bunch of mac addresses into a text box and send it to the backend
> 2. Backend needs to resolve the mac address to get the IP address
> 
> Problem: It's not possible to "scan a single mac address". To get the IP address from a mac you would need to scan a network range and then match the found mac's to the given ones. So you are back at this point:
> 
> > Using the built-in scan in UpSnap is not feasable, I just takes forever and generates a lot of noise in the network with the large number of IP's.
> 
> A possible solution would be to make both the mac and ip address required when batch importing. But that seems unintuitive and also is the website supposed to look like?
> 
> I haven't found a good solution to this yet.

Would uploading a list of IP's help? Or does it have to be mac-addresses? 

My current file has both Mac and IP, since it's a dnsmasq file.

**Reply by seriousm4x** [reply]
Just did some research if thats possible but unfortunately not.
It doesn't matter if you want to go from ip -> mac or mac -> ip. You always need to do an arp scan of your network.

**Reply by gaby** [reply]
Makes sense, thanks for the effort! 💪

**Reply by seriousm4x** [reply]
Maybe you can create a little bash script to read the dnsmasq file and make use of the api.

- https://pocketbase.io/docs/api-records/#create-record
- And take a look at the collection api in upsnaps backend:
http://localhost:8090/_/#/collections?collection=z5lghx2r3tm45n1 and click "API Preview" on the top right

**Reply by gaby** [reply]
Thanks, I will investigate.
